### DISCUSSION TITLE
How can I disable/enable RDNS for proxies?

### DISCUSSION THREAD

**Discussion by nsde** [encode/httpx#2812] [General]
E.g. in python-requests, I'd use `socks5h://`...
Is there any way I can do that using `httpx`?

I tried resolving just the IP itself and adding the Host header, but this gave me issues for Cloudflare servers and is a really insecure workaround.

Can anyone help me, please?

**Comment by lovelydinosaur** [comment]
See... https://github.com/encode/httpcore/discussions/673

I've not taken a proper look at the https://github.com/romis2012/httpx-socks package but perhaps this supports both `socks5` and `socks5h` forms? (Prompt: anyone want to add this to our third-party packages docs?)

If you'd like to get support for this added then we'd be able to make the time to work with you on it.

**Reply by nsde** [reply]
I'd love to see support for enabling/disabling `rdns`, because my proxy has issues with hostname resolution.
The person who opened the `httpcore` issue also started a PR, but it was closed: https://github.com/encode/httpcore/pull/672

**Reply by lovelydinosaur** [reply]
Yes, although it doesn't look to me like that PR *actually* implemented support.
There's several test cases, but the only behavioural change was including `socks5h` as an allowable URL protocol.

**Reply by nsde** [reply]
Alright, you could take a look at how `aiohttp-socks` implements `rdns`:
https://github.com/romis2012/aiohttp-socks/blob/1fafaf3e0db858088044b86ad65b4e2ab814d96b/aiohttp_socks/connector.py

**Reply by nsde** [reply]
I found that my proxy worked when I used `aiohttp-socks` and disabled `rdns`. (Which lead me to the creation of this discussion)

**Reply by nsde** [reply]
I did some more research and found out that `requests` uses `urllib3` for the `socks5h` functionality: https://github.com/urllib3/urllib3/blob/0e2c3a392d4112a66611dcdbac3e34e2f6378db0/src/urllib3/contrib/socks.py#L207

**Reply by lovelydinosaur** [reply]
So... two things here:

* Have you tried https://github.com/romis2012/httpx-socks?
* Would you like to resolve this directly in `httpx`/`httpcore`? If so I'd be able to guide you through it.

**Reply by nsde** [reply]
- No, but I'm going to try this
- Sure

**Comment by tyilo** [comment]
As I understand it `httpx` currently treats a `socks5://` url like a `socks5h://` one.

If there is no progress in adding both `socks5://` and `socks5h://` support, I think it would make sense to remove support for `socks5://` urls and only support `socks5h://` urls, instead of the current wrong implementation.
